db/change_cache.go

```diff
@@ -184,29 +183,34 @@ func (c *changeCache) Init(dbcontext *DatabaseContext, notifyChange func(base.Se
 
 	heap.Init(&c.pendingLogs)
 
-	// background tasks that perform housekeeping duties on the cache
-	bgt, err := NewBackgroundTask("InsertPendingEntries", c.context.Name, c.InsertPendingEntries, c.options.CachePendingSeqMaxWait/2, c.terminator)
-	if err != nil {
-		return err
-	}
-	c.backgroundTasks = append(c.backgroundTasks, bgt)
-
-	bgt, err = NewBackgroundTask("CleanSkippedSequenceQueue", c.context.Name, c.CleanSkippedSequenceQueue, c.options.CacheSkippedSeqMaxWait/2, c.terminator)
-	if err != nil {
-		return err
-	}
-	c.backgroundTasks = append(c.backgroundTasks, bgt)
+	go func() {
+		dcpStartedErr := <-dcpStarted
+		// TODO: how to test this error condition?
+		if dcpStartedErr != nil {
+			// error is handled in StartGOCB2DCPFeed
+			return
+		}
+		// background tasks that perform housekeeping duties on the cache
+		bgt, err := NewBackgroundTask("InsertPendingEntries", c.context.Name, c.InsertPendingEntries, c.options.CachePendingSeqMaxWait/2, c.terminator)
+		// TODO: how to test this error condition?
+		if err != nil {
+			base.ErrorfCtx(context.TODO(), "Failed to initialize task %w", err)
+			dbcontext.Close()
```

Comment:
Does this need to return, to stop a nil `bgt` being added to the `c.backgroundTasks` slice? It seems we're in a state that we can't continue from, but we are continuing regardless.

Reply:
Added return for clarity.